### ISSUE TITLE
Check template contents are valid for parent

### ISSUE THREAD

**Issue by MeirionHughes** [MeirionHughes/aurelia-template-lint#142]
given a html container,
when it has a `<template>` child
then ensure that the template has a template-controller
and ensure contents are valid for the html container. 

example: 

```html
<select>
  <template if.bind="foo" repeat.for="item of items">
     <option ... ></option> <!-- OK -->
     <div ...></div> <!-- INVALID -->
  </template>
</select>
```
should report that `<div>` is not a valid child of `<select>`

**Comment by zakjan** [comment]
+1

Just a note, if.bind and repeat.for are not allowed on the same element, because their order is undefined. They need to be nested.

**Comment by MeirionHughes** [comment]
Good point; It'll have to check through nested templates.
